### PR TITLE
Add ADR for "Use poetry for python venv management"

### PR DESCRIPTION
**Pull request by rpdelaney**
# [OHSH-578](https://ocio-jira.acf.hhs.gov/browse/OHSH-578)

Changes proposed in this pull request:

- Add ADR 0017: Use poetry for python venv management

### CODE REVIEW

File: docs/adr/0017-use-poetry-for-python-env-management.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+- Bad, because Truss have not used flit before.
+
+### poetry
+
+`poetry` is the holy marriage of pipenv, setuptools, and twine. It is a one-stop shop for python
+app development.
+
+```console
+# Install poetry
+$ brew install poetry
```
Comment (clint-truss) [inline docs/adr/0017-use-poetry-for-python-env-management.md:200]: Wow, I've never used poetry before. The closest I've used is pyenv, which seems close, but not nearly as user friendly. Does poetry run on old ubuntu systems as well?

**Review by clint-truss — APPROVED** [review]
I'm happy to trust to poetry right now, it sounds amazing, would like to know if it runs on old systems since we have a plethora of those. 

Other question, which I'm assuming is a 'yes' -- does it also work with python 3 and python 2 (in the unlikely and unfortunate event we have to do something in python 2 temporarily).

File: docs/adr/0017-use-poetry-for-python-env-management.md (hunk 1 above)
Comment (clint-truss) [reply docs/adr/0017-use-poetry-for-python-env-management.md:200]: Ah, this seems to answer all my questions: [https://python-poetry.org/docs/](https://python-poetry.org/docs/)

**Comment by rpdelaney** [comment]
We won't need to run poetry in production. It's only for making development more effective. Which version of the interpreter to target is definitely a legitimate concern for sure, but that is not addressed in this ADR.  Does that make sense?

**Comment by rpdelaney** [comment]
Forgot the other question: I haven't tried it with python 2. but I can look into it when I'm back at my desk

**Review by reid-truss — APPROVED** [review]

**Pull request merged**